subsys/bluetooth/host/keys.c

```diff
@@ -51,11 +59,32 @@ struct bt_keys *bt_keys_get_addr(u8_t id, const bt_addr_le_t *addr)
 			first_free_slot = i;
 		}
 	}
-
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (first_free_slot == ARRAY_SIZE(key_pool)) {
+		size_t oldest_slot = 0;
+
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			if (key_pool[oldest_slot].aging_counter
+			    > key_pool[i].aging_counter) {
+				oldest_slot = i;
+			}
+		}
+		bt_unpair(key_pool[oldest_slot].id,
+			  &key_pool[oldest_slot].addr);
+		if (!bt_addr_le_cmp(&key_pool[oldest_slot].addr,
+				    BT_ADDR_LE_ANY)) {
+			first_free_slot = oldest_slot;
+		}
+	}
+#endif  /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
 	if (first_free_slot < ARRAY_SIZE(key_pool)) {
 		keys = &key_pool[first_free_slot];
 		keys->id = id;
 		bt_addr_le_copy(&keys->addr, addr);
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+		keys->aging_counter = ++aging_counter_val;
+		last_keys_updated = keys;
+#endif          /* CONFIG_BT_KEYS_OVERWRITE_OLDEST */
```

Comment:
Delete this now? Since we have it in bt_key_notify_usage?

Reply:
This will initialize aging counter in compatibility mode. Can move this to compatibility mode parser.

Reply:
Found out that we need it because `bt_key_notify_usage` (reanmed into `bt_key_notify_usage`) is called after settings are save. So we need a proper counter value to be set into newly allocated storage slot